### PR TITLE
protocol changes for conflict resolution

### PR DESCRIPTION
**Pull request by songgao**
cc @jakob223 @strib

### CODE REVIEW

File: protocol/avdl/keybase1/favorite.avdl [hunk 1]
```diff
@@ -39,8 +63,7 @@ protocol favorite {
 
     union { null, Time } mtime;
 
-    // Is the local view of this folder hosed due to stuck CR?
-    FolderConflictType conflictType;
+    union { null, ConflictState } conflictState;
```
Comment (strib) [inline protocol/avdl/keybase1/favorite.avdl]: Remove the definition of `FolderConflictType` above?  Or I guess not until the rest of the Go code is fixed up to not depend on it.

File: protocol/avdl/keybase1/simple_fs.avdl [hunk 2]
```diff
@@ -370,11 +370,23 @@ protocol SimpleFS {
...
-  /**
-   Clear the conflict state of a TLF.
-   */
+  // Clear the conflict state of a TLF. This turns a locally "branched" TLF
+  // into:
+  //   1) a TLF that's as seen on all other devices, i.e. the server view,
+  //      under the normal TLF path;
+  //   2) a TLF that's locally branched, under a path with a TLF suffix.
+  //
+  // After this, user is expected to make change in 1), usually by coping needed
```
Comment (strib) [inline protocol/avdl/keybase1/simple_fs.avdl]: "change" -> "changes" or "some change"

File: go/kbfs/libkbfs/kbfs_ops.go [hunk 3]
```diff
@@ -314,7 +314,13 @@ func (fs *KBFSOpsStandard) GetFavoritesAll(ctx context.Context) (
 		if err != nil {
 			return keybase1.FavoritesResult{}, err
 		}
-		favs.FavoriteFolders[i].ConflictType = status
+		if status != keybase1.FolderConflictType_NONE {
+			stk := status == keybase1.FolderConflictType_IN_CONFLICT_AND_STUCK
+			conflictState := keybase1.NewConflictStateWithAutomaticresolving(
+				keybase1.ConflictAutomaticResolving{IsStuck: stk})
+			favs.FavoriteFolders[i].ConflictState = &conflictState
+
```
Comment (strib) [inline go/kbfs/libkbfs/kbfs_ops.go]: Extra blank line.

File: protocol/avdl/keybase1/favorite.avdl (hunk 1 above)
Comment (songgao) [reply protocol/avdl/keybase1/favorite.avdl]: Yeah, I think KBFSOps still returns that at this time.

File: go/kbfs/libkbfs/kbfs_ops.go (hunk 3 above)
Comment (songgao) [reply go/kbfs/libkbfs/kbfs_ops.go]: `padding-bottom` 😜

**Pull request merged**